### DISCUSSION TITLE
LPS node configuration issue in VM.

### DISCUSSION THREAD

**Discussion by Rishi23456** [bitcraze/discussions#1202] [Q&A]
Hello everyone, updating the firmware on the loco-positioning node has been an issue since a week. I am using the VM provided by the bitcraze, I have cloned the [Loco Positioning configuration tool GIT repo](https://github.com/bitcraze/lps-tools) and followed the [readme instruction](https://github.com/bitcraze/lps-tools#running-instruction) and gave the USB access rights on LINUX, yet after opening the LPS configuration tool, the tool is able to recognize the node after browsing the firmware and clicking on update. The firmware is not getting updated, instead the percentage bar shows 0% and the update button is greyed out and the cursor automatically goes on to configure the node ID: 

I have checked [Issue with updating LPS Node](https://forum.bitcraze.io/viewtopic.php?p=24378#p24378) this forum and tried all the debugging options mentioned. VM is able to recognize the node but not able to update any firmware into it.

Any reply would be of a great help. Thank You.

**Comment by ataffanel** [comment]
Can you copy-paste messages displayed in the console when you try to flash the node? If there is errors they should be displayed there.

The node will also disconnect and reconnect after being flashed, it could help to add both USB VID/PID for the firmware and the bootloader mode in the virtual machine USB filter list so that the devices connects automatically to the VM.

**Reply by Rishi23456** [reply]
Thanks for the reply Arnaud, 
The below pic is when i insert node in DFU mode.
<img width="593" alt="Screenshot 2024-03-04 142904" src="https://github.com/bitcraze/discussions/assets/107744277/e94141f0-c5ad-4b3c-882e-ec2672d0208b">
The Configuration tool is able to recognize the node, the below pic is when I select the lps-node-firmware.
<img width="573" alt="Screenshot 2024-03-04 145017" src="https://github.com/bitcraze/discussions/assets/107744277/57adb42d-9305-4c94-8483-58148bc76c34">
Now, if I click on update, as i mentioned cursor automatically goes on to id, without showing me prompt "Success" as supposed to and the node too doesn't restart instead stays in the DFU mode.
<img width="581" alt="LPS Update" src="https://github.com/bitcraze/discussions/assets/107744277/2b51bdac-f8a5-4e2b-b915-eac1801c325d">

**Reply by ataffanel** [reply]
Hi, thanks for the picture, it indeed does not look good. There is a warning in the console on the back, what is the warning saying?

**Reply by Rishi23456** [reply]
<img width="368" alt="Warning" src="https://github.com/bitcraze/discussions/assets/107744277/877a1550-f4c2-4a36-9a5a-9f705b5480fc">

**Reply by ataffanel** [reply]
I am not able to reproduce the problem right now unfortunatly, Can you try to run natively outside of the VM to see if it works there?

Looking at the LPS tools it does need quite a refresh, I am adding a couple of ticket in the repos and we will look at it in the following weeks to try to get it in a better state.

We have seen a lot of problems lately with the VM, the USB subsystem of virtualbox seems to work much worst than before, this is also something we need to investigate, so in the mean time the best would be to work on your machine's python directly instead.

**Comment by Rishi23456** [comment]
Can we only install the LPS configuration tool Windows Native installer
<https://github.com/bitcraze/lps-tools/releases> for windows and can update
the firmware to the LOCO-node or do we need to install the Crazy flie
client too.

ThankYou

**Comment by Rishi23456** [comment]
The Loco Nodes got Updated on the Windows version on my laptop.
